sdk/eventgrid/eventgrid/review/eventgrid.api.md

```diff
@@ -0,0 +1,206 @@
+}
+
+// @public
+export type ContainerRegistryChartDeletedEventData = ContainerRegistryArtifactEventData & {};
+
+// @public
+export type ContainerRegistryChartPushedEventData = ContainerRegistryArtifactEventData & {};
+
+// @public
+export interface ContainerRegistryEventActor {
+    name?: string;
+}
+
+// @public
+export interface ContainerRegistryEventData {
+    action?: string;
+    actor?: ContainerRegistryEventActor;
+    id?: string;
+    request?: ContainerRegistryEventRequest;
+    source?: ContainerRegistryEventSource;
+    target?: ContainerRegistryEventTarget;
+    timestamp?: Date;
+}
+
+// @public
+export interface ContainerRegistryEventRequest {
+    addr?: string;
+    host?: string;
+    id?: string;
+    method?: string;
+    useragent?: string;
+}
+
+// @public
+export interface ContainerRegistryEventSource {
+    addr?: string;
+    instanceID?: string;
+}
+
+// @public
+export interface ContainerRegistryEventTarget {
+    digest?: string;
+    length?: number;
+    mediaType?: string;
+    repository?: string;
+    size?: number;
+    tag?: string;
+    url?: string;
+}
+
+// @public
+export type ContainerRegistryImageDeletedEventData = ContainerRegistryEventData & {};
+
+// @public
+export type ContainerRegistryImagePushedEventData = ContainerRegistryEventData & {};
+
+// @public
+export class EventGridClient {
+    constructor(endpointUrl: string, credential: KeyCredential | SignatureCredential, options?: EventGridClientOptions);
+    readonly apiVersion: string;
+    readonly endpointUrl: string;
+    generateSharedAccessSignature(expiresOnUtc: Date): string;
+    sendCloudEvents(events: CloudEvent<any>[], options?: SendCloudEventsOptions): Promise<RestResponse>;
+    sendCustomSchemaEvents(events: Record<string, any>[], options?: SendCustomSchemaEventsOptions): Promise<RestResponse>;
+    sendEvents(events: EventGridEvent<any>[], options?: SendEventsOptions): Promise<RestResponse>;
+}
+
+// @public
+export type EventGridClientOptions = PipelineOptions;
+
+// @public
+export class EventGridConsumer {
+    constructor(decoders?: Record<string, CustomEventDataDecoder>);
+    // Warning: (ae-forgotten-export) The symbol "CustomEventDataDecoder" needs to be exported by the entry point index.d.ts
+    //
+    // (undocumented)
+    readonly customDecoders: Record<string, CustomEventDataDecoder>;
+    decodeCloudEvents(encodedEvents: string): Promise<CloudEvent<unknown>[]>;
+    decodeCloudEvents(encodedEvents: object): Promise<CloudEvent<unknown>[]>;
+    decodeEventGridEvents(encodedEvents: string): Promise<EventGridEvent<unknown>[]>;
+    decodeEventGridEvents(encodedEvents: object): Promise<EventGridEvent<unknown>[]>;
+}
+
+// @public
+export interface EventGridEvent<T> {
+    data: T;
+    dataVersion: string;
+    eventTime: Date;
+    eventType: string;
+    id: string;
+    subject: string;
+    topic?: string;
+}
+
+// @public
+export class EventGridSharedAccessSignatureCredential implements SignatureCredential {
```

Comment:
/cc @schaabs, our credential expert, in case he hasn't seen this yet

Reply:
FWIW, Scott and I did talk about this a bit and this does follow the pattern we sort of came up with.  The key feedback from Scott was he didn't want one of these things to be able to be passed to something that would expect an `KeyCredential`, which is why the shapes of the interfaces are slightly different.